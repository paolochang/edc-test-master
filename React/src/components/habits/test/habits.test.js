import React from "react";
import { render, screen } from "@testing-library/react";
import Habits from "../habits";
import userEvent from "@testing-library/user-event";

describe("Habits", () => {
  const habits = [
    { id: 1, name: "Reading", count: 1 },
    { id: 2, name: "Running", count: 0 },
  ];
  let component;
  let onReset;

  it("compares snapshot", () => {
    component = render(
      <Habits
        habits={habits}
        onIncrement={jest.fn}
        onDecrement={jest.fn}
        onDelete={jest.fn}
        onAdd={jest.fn}
        onReset={jest.fn}
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  describe("tests user interface", () => {
    beforeEach(() => {
      onReset = jest.fn();
      render(
        <Habits
          habits={habits}
          onIncrement={jest.fn}
          onDecrement={jest.fn}
          onDelete={jest.fn}
          onAdd={jest.fn}
          onReset={onReset}
        />
      );
    });

    it("calls reset when reset button is clicked", () => {
      const btnReset = screen.getByText("Reset All");
      userEvent.click(btnReset);
      expect(onReset).toHaveBeenCalledTimes(1);
    });
  });
});
