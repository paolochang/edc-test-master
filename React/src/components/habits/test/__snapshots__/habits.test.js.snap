// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Habits compares snapshot 1`] = `
<div>
  <div
    class="habits"
  >
    <form
      class="add-form"
    >
      <input
        class="add-input"
        placeholder="Habit"
        type="text"
      />
      <button
        class="add-button"
      >
        Add
      </button>
    </form>
    <ul>
      <li
        class="habit"
      >
        <span
          class="habit-name"
        >
          Reading
        </span>
        <span
          class="habit-count"
        >
          1
        </span>
        <button
          class="habit-button habit-increase"
          title="increase"
        >
          <i
            class="fas fa-plus-square"
          />
        </button>
        <button
          class="habit-button habit-decrease"
          title="decrease"
        >
          <i
            class="fas fa-minus-square"
          />
        </button>
        <button
          class="habit-button habit-delete"
          title="delete"
        >
          <i
            class="fas fa-trash"
          />
        </button>
      </li>
      <li
        class="habit"
      >
        <span
          class="habit-name"
        >
          Running
        </span>
        <span
          class="habit-count"
        >
          0
        </span>
        <button
          class="habit-button habit-increase"
          title="increase"
        >
          <i
            class="fas fa-plus-square"
          />
        </button>
        <button
          class="habit-button habit-decrease"
          title="decrease"
        >
          <i
            class="fas fa-minus-square"
          />
        </button>
        <button
          class="habit-button habit-delete"
          title="delete"
        >
          <i
            class="fas fa-trash"
          />
        </button>
      </li>
    </ul>
    <button
      class="habits-reset"
    >
      Reset All
    </button>
  </div>
</div>
`;
